src/test/java/org/example/stepsDefinitions/LoginFunctionalitiesSteps.java

```diff
@@ -0,0 +1,30 @@
+package org.example.stepsDefinitions;
+
+import io.cucumber.java.en.*;
+
+
+public class LoginFunctionalitiesSteps {
+	@Given("the user has navigated to the login page")
+	public void theUserHasNavigatedToTheLoginPage() {
+
+	}
+
+	@When("the user enters the correct username and password")
+	public void theUserEntersTheCorrectUsernameAndPassword() {
+
+	}
+
+	@Then("the user should be redirected to the home page")
+	public void theUserShouldBeRedirectedToTheHomePage() {
+		
+	}
+
+	@When("the user enters an incorrect username or password")
+	public void theUserEntersAnIncorrectUsernameOrPassword() {
+		
+	}
+
+	@Then("the user should see an error message indicating invalid login credentials")
```

Comment:
The error message appears with {string} context.
provide expected context as an string param to the method.
With this approach the step will be more flexible to different proposes and can be used in different tests.

Reply:
Error Message validation implemented!